docs/guides/checkpointing_solutions/GCS_Checkpointing.md

```diff
@@ -0,0 +1,113 @@
+# GCS bucket-based checkpointing
+
+In this type of checkpointing, the checkpoints are saved directly to a Google
+Cloud Storage (GCS) bucket. This is the most straightforward and common method.
+The training process briefly pauses, for capturing the model's state and keeping
+the training in sync with rest of the workers, then the model's state is
+asynchronously serialized and written over the network to the specified GCS
+bucket.
+
+### Checkpoint Loading Priority
+
+The system follows a specific order when deciding which checkpoint to load at startup. The first valid condition met is the one executed:
+
+1.  **Resume Current Run**: If a checkpoint already exists for the current `run_name`, the system loads the latest full checkpoint. This is the default behavior to ensure minimal state loss when resuming after an interruption.
+2.  **Load from Specific Path**: If no local checkpoint is found, the system checks for a user-specified path.
+    * If `load_parameters_path` is set, it loads **only the model parameters**.
+    * If `load_full_state_path` is set, it loads the **entire training state** (parameters, optimizer state, step count, etc.).
+    * **Note**: These two options are mutually exclusive and will cause an error if both are set.
+3.  **Initialize from Scratch**: If none of the above conditions are met, the model initializes with new weights.
+
+---
+
+### Maxtext Configuration
+
+#### `enable_checkpointing`
```

Comment:
Do we really want to repeat these flags here? Maybe we want to document them once in a parent document (which will likely be needed anyway), and for each checkpointing type we only need to mention which flags are relevant? 

In any case, my suggestion is again to use a table or a bullet-list.